Update to the current ws or deprecate?

ChALkeR opened the pull request:
Judging from the npm numbers, this fork ([ws-pure](https://www.npmjs.com/package/ws-pure)) has about 3% of the total `ws` downloads, which is 122000 downloads per month, and that number is quite high.

This fork does not seem to be up to date with recent security fixes in `ws`, namely 29293ed11b679e0366fa0f6bb9310b330dafd795, so anyone who uses this fork is still affected. See https://nodesecurity.io/advisories/67 for details.

Note that there are no patches on top of `ws` now in this fork, and `ws` itself doesn't depend on native modules anymore afaik.

So, two possible ways to fix this:
1. Keep this up-to-date with upstream `ws`, and keep up with the security fixes.
2. Deprecate this module and replace it in npm registrtry with a small wrapper that depends on upstream `ws`, re-exports it, and prints a deprecation warning.

Commit:
added small fix to prevent server from crashing on empty this.state

jondubois committed:
Status code must be 1006 if the socket is closed without having received a close control frame

Commit:
Fixed error emission on receiver error

Commit:
Added type check before receiver error emission

Commit:
Added unit tests for receiver error emission

nkzawa committed:
don't reuse buffer

sebpiq committed:
Clean event handlers on WebSocketServer's internal http server when closing

darrachequesne committed:
Release memory used by PerMessageDeflate extension

akyoto committed:
Callback for close()

This will make close() compatible with Promise libraries by providing a callback when its finished.
Note that this version only works correctly for the cases where the http server was **not** internally created.
If you like, you can support the other case as well (would need to pass a callback to `this._closeServer()`).

3rd-Eden committed:
Merge pull request #608 from darrachequesne/master

Release memory used by PerMessageDeflate extension

3rd-Eden committed:
Merge pull request #611 from blitzprog/patch-1

Callback for close() on WebSocketServer

3rd-Eden committed:
Merge pull request #595 from nkzawa/patch-5

Fix error when permessage-deflate is enabled

3rd-Eden committed:
[dist] 0.8.1

Tatsujinichi committed:
fix for issue 603

darrachequesne committed:
Fix an error on Node v0.8 (no 'close' method for InflateRaw/DeflateRaw)

3rd-Eden committed:
Merge pull request #625 from darrachequesne/patch-1

Fix an error on Node v0.8 (no 'close' method for InflateRaw/DeflateRaw)

darrachequesne committed:
Fix "zlib binding closed" errors

3rd-Eden committed:
Merge pull request #623 from Tatsujinichi/issue603

fix for issue 603

SEAPUNK committed:
Sender: Don't reuse masking key

3rd-Eden committed:
Merge pull request #632 from darrachequesne/patch-2

Fix "zlib binding closed" errors

3rd-Eden committed:
[major] No longer use the binary addons as optional dependency, nuked completely

3rd-Eden committed:
[breaking] Remove browser code. Out of scope for this module.

3rd-Eden committed:
Merge pull request #566 from SocketCluster/master

Status code must be 1006 if the socket is closed without having recei…

3rd-Eden committed:
[pkg] Bump dependencies

3rd-Eden committed:
[travis] Remove deprecated node.js version 0.12 is the last to be supported

dignifiedquire committed:
fix: shrinkStrategy should always return an integer

`Buffer` inherits from `Uint8Arry` so the constructor needs to be
called with integers and mustn't be called with floats.

This patch ensures that the result of the the calculation
in the `shrinkStrategy` in `Receiver` returns an integer.

The reason why this has not been a problem up to now is that
a there is a [V8 Bug](https://code.google.com/p/v8/issues/detail?id=4552)
that the [spec](http://tc39.github.io/ecma262/#sec-typedarray-length)
is not adhered in this case.

This is a blocking issue for [Karma](https://github.com/karma-runner/karma/issues/1768)
as the upgrade to [core-js@2.0.0](https://github.com/zloirock/core-js)
now enforces this error.

3rd-Eden committed:
Merge pull request #645 from Dignifiedquire/buffer-length

fix: shrinkStrategy should always return an integer

3rd-Eden committed:
[doc] Document the new optional dependencies

3rd-Eden committed:
[dist] 1.0.0

LordMajestros committed:
Add maxpayload limit to hybi clients

Limit the amount of data that can be sent to a websocketserver from a
hybi client.
Raise a 1009 - message too big error.
Tests included.

LordMajestros committed:
Add maxpayload limit to hybi clients

Limit the amount of data that can be sent to a websocketserver from a
hybi client.
Raise a 1009 - message too big error.
Tests included.

3rd-Eden committed:
[fix] Prevent allocation of memory through ping frames

3rd-Eden committed:
[dist] 1.0.1

3rd-Eden committed:
[security] Added a missing Security.md

JacksonTian committed:
remove the browserify users section

JacksonTian committed:
Add make coverage command to get test coverage info

JacksonTian committed:
Use Buffer.concat() instead of private concatBuffers method

JacksonTian committed:
Add strict rule and update code

calebboyd committed:
[pkg]: add `main` field to package.json

3rd-Eden committed:
Merge pull request #666 from calebboyd/master

[pkg]: add `main` field to package.json

LordMajestros committed:
Merge branch 'master' of https://github.com/LordMajestros/ws

Merge security patches into this pull request

3rd-Eden committed:
Merge pull request #650 from LordMajestros/master

Add maxpayload limit to hybi clients

3rd-Eden committed:
Merge pull request #652 from JacksonTian/browser

remove the browserify users section

johanneswuerbach commented:
The source of the module is actually https://github.com/testem/ws and they are no longer in use, I'll mark them as deprecated asap.

LordMajestros committed:
Fix for #680 Maxpayload error (https://github.com/websockets/ws/issues/680)

LordMajestros committed:
Fix for https://github.com/websockets/ws/issues/679 unhandled opcodes cause crash

LordMajestros committed:
Add fix to Receiver.hixie

agilethomas committed:
Fix for #687 Hixie-76 doesn't work behind HAProxy

- Solution is to send headers before receiving the nonce

josephfrazier committed:
[doc] Fix grammar of utf-8-validate info

3rd-Eden committed:
Merge pull request #697 from josephfrazier/patch-1

[doc] Fix grammar of utf-8-validate info

rib committed:
lib/WebSocket.js: expose binaryType property

The W3C WebSocket interface includes a binaryType attributes for a
socket and a corresponding BinaryType { 'blob', 'arraybuffer' } enum.
This enables support for the 'arraybuffer' BinaryType, but avoiding
changing the existing behaviour the default binaryType is set as
'nodebuffer'. The 'blob' type is left unsupported.

This improves ws compatibility with some existing W3C WebSocket based
client code.

Fixes: #441

wesleytodd committed:
added documentation for close callback

3rd-Eden committed:
Merge pull request #703 from StreamMeDev/close-doc-update

Added documentation for close callback

3rd-Eden committed:
Merge pull request #698 from rib/binarytype

lib/WebSocket.js: expose binaryType property

3rd-Eden committed:
Merge pull request #681 from LordMajestros/master

Fix for #680 (https://github.com/websockets/ws/issues/680)

misobytes committed:
emit both an error and close event when connection establishment fails

3rd-Eden committed:
Merge pull request #706 from misobytes/emitCloseEventOnConnectionFailure

Emit both an error and close event when connection establishment fails

3rd-Eden committed:
Merge pull request #605 from sebpiq/master

Clean event handlers on WebSocketServer's internal http server when closing

3rd-Eden committed:
Merge pull request #654 from JacksonTian/concat

Use Buffer.concat() instead of private concatBuffers method

3rd-Eden committed:
Merge pull request #688 from agilethomas/hixie76_haproxy

Fix for #687 Hixie-76 doesn't work behind HAProxy

3rd-Eden committed:
Merge pull request #569 from cybusio/master

Fixed error emission on receiver error

3rd-Eden committed:
Merge pull request #653 from JacksonTian/coverage

Add make coverage command to get test coverage info

3rd-Eden committed:
Merge pull request #636 from SEAPUNK/patch-1

Sender: Don't reuse masking key

3rd-Eden committed:
Merge pull request #336 from MosNicholas/nmos_server_crash_fix

added small fix to prevent server from crashing on empty this.state

3rd-Eden committed:
[dist] 1.1.0

mscdex committed:
test: update `should` usage

These changes are in anticipation of upcoming changes in node v7.0.0.
See: https://github.com/nodejs/node/pull/6102

3rd-Eden committed:
Merge pull request #720 from mscdex/test-fix-should-usage

test: update `should` usage

JacksonTian committed:
Clean up code

JacksonTian committed:
Add node 6 into travis list

JacksonTian committed:
Add eslint as lint tool

3rd-Eden committed:
Merge pull request #741 from JacksonTian/travis

Add node 6 into travis list

3rd-Eden committed:
Merge pull request #743 from JacksonTian/lint

Add eslint as lint tool

JacksonTian committed:
fix PerMessageDeflate options pass issue

JacksonTian committed:
refine code via eslint

tejasmanohar committed:
specify data encoding type 'binary' at hash#update

3rd-Eden committed:
Merge pull request #744 from JacksonTian/fix

fix PerMessageDeflate options pass issue

3rd-Eden committed:
Merge pull request #745 from JacksonTian/refine

refine code via eslint

3rd-Eden committed:
[ci] Nuke 0.12

3rd-Eden committed:
Merge pull request #746 from tejasmanohar/fix/node_6_crypto_binary

specify data encoding type 'binary' at hash#update

3rd-Eden committed:
Merge pull request #739 from JacksonTian/clean_up

Clean up code

JacksonTian committed:
Add quotes rule and fix coding style

JacksonTian committed:
add indent rule and update code

3rd-Eden committed:
Merge pull request #748 from JacksonTian/indent

add indent rule and update code

JacksonTian committed:
Add space-infix-ops rule and refine code (#747)

JacksonTian committed:
lint the project before run test

JacksonTian committed:
Merge pull request #655 from JacksonTian/use_strict

use strict mode

JacksonTian committed:
Merge pull request #749 from JacksonTian/quotes

Add quotes rule and fix coding style

JacksonTian committed:
Add keyword-spacing rule (#752)
